libraries/AP_Scripting/docs/docs.lua

```diff
@@ -1116,10 +1116,12 @@ function baro:get_pressure() end
 ---@class serial
 serial = {}
 
--- desc
----@param protocol integer
----@return AP_HAL__UARTDriver_ud
-function serial:find_serial(protocol) end
+-- Returns the UART instance that allows connections from scripts (those with SERIALx_PROTOCOL = 28`).
+-- For instance = 0, returns first such UART, second for instance = 1, and so on.
+-- If such an instance is not found, returns nil.
+---@param instance integer -- the 0-based index of the UART instance to return.
+---@return AP_HAL__UARTDriver_ud -- the requested UART instance available for scripting, or nil if none.
```

Comment:
You can also do this if its a null able return, I'n not sure why its not that already.... It should have been auto generated like that.....
```suggestion
---@return AP_HAL__UARTDriver_ud|nil -- the requested UART instance available for scripting, or nil if none.
```

Reply:
I've got an impression that `T|nil` is written only if T is one or more primitive types (counting `number` in). If one of the possible types is an "object" (such as string) then it's nullable by default. However, I don't know Lua habits.